Mathlib/Algebra/Order/Module/Defs.lean

```diff
@@ -0,0 +1,826 @@
+## Definitions
+
+In all that follows, `α` and `β` are orders which have a `0` and such that `α` acts on `β` by scalar
+multiplication. Note however that we do not use lawfulness of this action in most of the file. Hence
+`•` should be considered here as a mostly arbitrary function `α → β → β`.
+
+We use the following four typeclasses to reason about left scalar multiplication (`b ↦ a • b`):
+* `PosSMulMono`: If `a ≥ 0`, then `b₁ ≤ b₂` implies `a • b₁ ≤ a • b₂`.
+* `PosSMulStrictMono`: If `a > 0`, then `b₁ < b₂` implies `a • b₁ < a • b₂`.
+* `PosSMulReflectLT`: If `a ≥ 0`, then `a • b₁ < a • b₂` implies `b₁ < b₂`.
+* `PosSMulReflectLE`: If `a > 0`, then `a • b₁ ≤ a • b₂` implies `b₁ ≤ b₂`.
+
+We use the following four typeclasses to reason about right scalar multiplication (`a ↦ a • b`):
+* `SMulPosMono`: If `b ≥ 0`, then `a₁ ≤ a₂` implies `a₁ • b ≤ a₂ • b`.
+* `SMulPosStrictMono`: If `b > 0`, then `a₁ < a₂` implies `a₁ • b < a₂ • b`.
+* `SMulPosReflectLT`: If `b ≥ 0`, then `a₁ • b < a₂ • b` implies `a₁ < a₂`.
+* `SMulPosReflectLE`: If `b > 0`, then `a₁ • b ≤ a₂ • b` implies `a₁ ≤ a₂`.
+
+## Constructors
+
+The four typeclasses about nonnegativity can usually be checked only on positive inputs due to their
+condition becoming trivial when `a = 0` or `b = 0`. We therefore make the following constructors
+available: `PosSMulMono.of_pos`, `PosSMulReflectLT.of_pos`, `SMulPosMono.of_pos`,
+`SMulPosReflectLT.of_pos`
+
+## Implications
+
+As `α` and `β` get more and more structure, those typeclasses end up being equivalent. The commonly
+used implications are:
+* When `α`, `β` are partial orders:
+  * `PosSMulStrictMono → PosSMulMono`
+  * `SMulPosStrictMono → SMulPosMono`
+  * `PosSMulReflectLE → PosSMulReflectLT`
+  * `SMulPosReflectLE → SMulPosReflectLT`
+* When `β` is a linear order: `PosSMulStrictMono → PosSMulReflectLE`
+* When `α` is a linear order: `SMulPosStrictMono → SMulPosReflectLE`
+* When `α` is an ordered ring, `β` an ordered group and also an `α`-module:
+  * `PosSMulMono → SMulPosMono`
+  * `PosSMulStrictMono → SMulPosStrictMono`
+* When `α` is an ordered semifield, `β` is an `α`-module:
+  * `PosSMulStrictMono → PosSMulReflectLT`
+  * `PosSMulMono → PosSMulReflectLE`
+
+Further, the bundled non-granular typeclasses imply the granular ones like so:
+* `OrderedSMul → PosSMulStrictMono`
+* `OrderedSMul → PosSMulReflectLT`
+
+All these are registered as instances, which means that in practice you should not worry about these
+implications. However, if you encounter a case where you think a statement is true but not covered
+by the current implications, please bring it up on Zulip!
+
+## TODO
+
+This file acts as a substitute for `Mathlib.Algebra.Order.SMul`. We now need to
+* finish the transition by deleting the duplicate lemmas
+* rearrange the non-duplicate lemmas into new files
+* generalise (most of) the lemmas from `Mathlib.Algebra.Order.Module` to here
+* rethink `OrderedSMul`
+-/
+
+variable (α β : Type*)
+
+section Abbreviations
```

Comment:
There are no abbreviations here.

Can you add an implementation note that compares why we use dedicated classes here, but abbreviations for the `Mul` versions?

Reply:
There's not really any reason to use `abbrev` for the `Mul` versions, but "Let's keep the diff small".

Reply:
Copied over the blurb from my PR description